homeassistant/components/onvif/switch.py

```diff
@@ -21,88 +71,35 @@ async def async_setup_entry(
     """Set up a ONVIF switch platform."""
     device = hass.data[DOMAIN][config_entry.unique_id]
 
-    entities = [
-        ONVIFWiperSwitch(device),
-        ONVIFAutoFocusSwitch(device),
-        ONVIFInfraredSwitch(device),
-    ]
-    async_add_entities(entities)
-
-
-class ONVIFImagingSettingSwitch(ONVIFBaseEntity, SwitchEntity):
-    """An ONVIF switch controlled via ImagingSettings."""
-
-    _on_settings: dict[str, Any] = {}
-    _off_settings: dict[str, Any] = {}
-
-    async def async_turn_on(self, **kwargs) -> None:
-        """Turn on switch."""
-        self._attr_is_on = True
-        profile = self.device.profiles[0]
-        await self.device.async_set_imaging_settings(profile, self._on_settings)
+    async_add_entities([ONVIFSwitch(device, description) for description in SWITCHES])
 
-    async def async_turn_off(self, **kwargs) -> None:
-        """Turn off switch."""
-        self._attr_is_on = False
-        profile = self.device.profiles[0]
-        await self.device.async_set_imaging_settings(profile, self._off_settings)
 
+class ONVIFSwitch(ONVIFBaseEntity, SwitchEntity):
+    """An ONVIF switch."""
 
-class ONVIFAuxSwitch(ONVIFBaseEntity, SwitchEntity):
-    """An ONVIF switch controlled via ONVIF Auxiliary Commands."""
+    entity_description: ONVIFSwitchEntityDescription
 
-    _on_cmd = ""
-    _off_cmd = ""
+    def __init__(
+        self, device: ONVIFDevice, description: ONVIFSwitchEntityDescription
+    ) -> None:
+        """Initialize the switch."""
+        super().__init__(device)
+        self._attr_name = f"{self.device.name} {description.name}"
```

Comment:
Please set the class attribute `_attr_has_entity_name` to `True` instead and remove this line.

https://developers.home-assistant.io/docs/core/entity#entity-naming